herbert_core/utilities/general/range_add_border.m

```diff
@@ -0,0 +1,58 @@
+function range=range_add_border(range_in, tol)
+% Add a small border to a range.
+%
+%   >> range=range_add_border(range_in)
+%   >> range=range_add_border(range_in,tol)
+%
+% Input:
+% ------
+%   range_in   Range of data (2xn array)
+%               [u1_min,u2_min,...;u1_max,u2_max,...]
+%   if tol is omitted, the tol assumed to be equal -eps where eps
+%   is minimal value such as 1+eps~=1;
+%
+%   tol         Control size of border:
+%               tol=0   No border
+%               tol>0   Absolute value of thickness of border
+%               tol<0   Relative size as a proportion of the range along
+%                       each axis. If the range is zero, absolute tol value
+%                       is used.
+
+% Output:
+% -------
+%   range      Expanded range
+%
+%
+if nargin == 1 % add epsilon-sized border
+    tol = -eps;
+end
+%
+ndim=size(range_in,2);
+if tol==0
+    range=range_in;
+    return
+elseif tol>0
+    range=range_in+tol*([-ones(1,ndim);ones(1,ndim)]);
+    zero_width = abs(range_in(2,:)-range_in(1,:))<tol;
+    if any(zero_width) % abserr for zero-width border is defined as relerr
+        sig_range = sign(range_in);
+        min_border = 1-tol*sig_range(1,:);
+        max_border = 1+tol*sig_range(2,:);
+        range(1,zero_width)=range(1,zero_width).*min_border(zero_width);
```

Comment:
This is very clever code and I had to walk through it in the debugger to find out it was working. A description of how it works would save a future developer some time when debugging.

Reply:
difficult to explain what the code explain the best but I have tried. At least I've expressed intentions.